server/container/util/src/main/java/org/apache/james/util/ReactorUtils.java

```diff
@@ -18,10 +18,80 @@
  ****************************************************************/
 package org.apache.james.util;
 
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+import java.util.Spliterator;
+import java.util.stream.Stream;
+
+import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 public class ReactorUtils {
     public static <T> Mono<T> executeAndEmpty(Runnable runnable) {
         return Mono.fromRunnable(runnable).then(Mono.empty());
     }
+
+    public static InputStream toInputStream(Flux<byte[]> byteArrays) {
+        return new StreamInputStream(byteArrays.toStream(1));
+    }
+
+    private static  class StreamInputStream extends InputStream {
+        private static final int NO_MORE_DATA = -1;
+
+        private final Stream<byte[]> source;
```

Comment:
a little comment, how do you think if having a `Stream<ByteBuffer>` to avoid some intermediate transformation from cassandra row -> bytebuffer -> byte[] -> ByteArrayInputStream ?